### PR TITLE
Add support for React 16

### PR DESCRIPTION
**Pull request by joshwcomeau**
With React's official launch, I figured it was time to update the peer dependencies!

While trying to publish, I ran into some trouble; it wasn't able to find `babel-cli` when running the `lib` build step :/

Upgrading the dependency seems to fix it (publish still failed, but only because I hadn't yet incremented the version). I figured before I published, though, I should get another pair of eyes.

Mainly I'm curious if there's a more-ideal way to add support for React 16 in the peer dependencies.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "react-flip-move",
-  "version": "2.9.14",
```
Comment (joshwcomeau) [inline package.json]: Guess I never pushed the `2.9.15` branch, my bad. That was the release for zombie-nodes bugfix.

**Comment by Hypnosphi** [comment]
Did you test it with components returning strings or fragments?

**Comment by Hypnosphi** [comment]
> a more-ideal way to add support for React 16 in the peer dependencies

How about wildcard?

File: package.json [hunk 2]
```diff
@@ -46,14 +46,14 @@
     "web-animations"
   ],
   "peerDependencies": {
-    "react": "0.13.x || 0.14.x || 15.x.x",
-    "react-dom": "0.13.x || 0.14.x || 15.x.x"
+    "react": "0.13.x || 0.14.x || 15.x.x || 16.x.x",
```
Comment (tobilen) [inline package.json]: would `"react": ">=0.13.x <=16.x.x",` work?

**Comment by Hypnosphi** [comment]
> `if (typeof this.props.children === 'string' || typeof this.props.children === 'array')`

It's not about `this.props.children`, it's about output of each child's `render`

**Comment by tobilen** [comment]
check is not needed anymore when we merge https://github.com/joshwcomeau/react-flip-move/pull/191

**Comment by joshwcomeau** [comment]
This is no longer needed thanks to @tobilen's fantastic work :D

**Pull request closed without merging**